lib/wallet/integration/src/Test/Integration/Scenario/API/Shared/Transactions.hs

```diff
@@ -1608,6 +1617,187 @@ spec = describe "SHARED_TRANSACTIONS" $ do
         r <- request @(ApiTransaction n) ctx link Default Empty
         expectResponseCode HTTP.status404 r
         expectErrorMessage (errMsg404CannotFindTx $ toText txid2) r
+
+    it "SHARED_TRANSACTIONS_DELEGATION_01 - \
+        \Cannot delegate when wallet is missing a delegation script template" $
+        \ctx -> runResourceT $ do
+
+        (ApiSharedWallet (Right w)) <- emptySharedWallet ctx
+
+        pool1:_ <- map (view $ _Unwrapped . #id) . snd <$>
+            unsafeRequest @[ApiT StakePool]
+            ctx (Link.listStakePools arbitraryStake) Empty
+
+        let delegationJoin = Json [json|{
+                "delegations": [{
+                    "join": {
+                        "pool": #{ApiT pool1},
+                        "stake_key_index": "0H"
+                    }
+                }]
+            }|]
+        rTx <- request @(ApiConstructTransaction n) ctx
+            (Link.createUnsignedTransaction @'Shared w) Default delegationJoin
+        expectResponseCode HTTP.status403 rTx
+        decodeErrorInfo rTx `shouldBe` StakingInvalid
+
+    it "SHARED_TRANSACTIONS_DELEGATION_01a - \
+       \Can join stakepool, rejoin another and quit" $ \ctx -> runResourceT $ do
```

Comment:
Integration tests only covers `join` atm. `rejoin` and `quit` are missing if I'm not mistaken.

Also, I've noticed (after `joining` shared wallet on preprod) that the delegation settings returned as part of [getSharedWallet](https://input-output-hk.github.io/cardano-wallet/api/edge/#operation/getSharedWallet) are not updated and still show:
```
  "delegation": {
    "active": {
      "status": "not_delegating"
    },
    "next": []
  }
```
Therefore we need to make sure to also check something like [this](https://github.com/input-output-hk/cardano-wallet/blob/master/lib/wallet/integration/src/Test/Integration/Scenario/API/Shelley/StakePools.hs#L577-L583).